src/test/java/net/snowflake/client/jdbc/SnowflakeUtilTest.java

```diff
@@ -46,6 +48,40 @@ public void testCreateMetadata() throws Throwable {
         OBJECT_MAPPER.writeValueAsString(columnMetadata));
   }
 
+  @Test
+  public void testCreateFieldsMetadataForObject() throws Throwable {
+    // given
+    ObjectNode rootNode = createRootNode();
+    ArrayNode fields = OBJECT_MAPPER.createArrayNode();
+    fields.add(
+        OBJECT_MAPPER.readTree(
+            "{\"fieldName\":\"name1\", \"fieldType\": {\"type\":\"text\",\"precision\":null,\"byteLength\":256,\"scale\":null,\"nullable\":false,\"collation\":\"collation\",\"length\":256}}"));
+    fields.add(
+        OBJECT_MAPPER.readTree(
+            "{\"fieldName\":\"name2\", \"fieldType\": {\"type\":\"real\",\"precision\":5,\"byteLength\":128,\"scale\":null,\"nullable\":true,\"collation\":\"collation\",\"length\":256}}"));
+    rootNode.putIfAbsent("fields", fields);
+
+    // when
+    SnowflakeColumnMetadata columnMetadata =
+        SnowflakeUtil.extractColumnMetadata(rootNode, false, null);
+    // then
+    assertNotNull(columnMetadata);
+    assertEquals("OBJECT", columnMetadata.getTypeName());
+
+    FieldMetadata firstField = columnMetadata.getFields().get(0);
+    assertEquals("name1", firstField.getName());
+    assertEquals(SnowflakeType.TEXT, firstField.getBase());
+    assertEquals(256, firstField.getByteLength());
+    assertFalse(firstField.isNullable());
+
+    FieldMetadata secondField = columnMetadata.getFields().get(1);
+    assertEquals("name2", secondField.getName());
+    assertEquals(SnowflakeType.REAL, secondField.getBase());
+    assertEquals(256, secondField.getByteLength());
```

Comment:
second byte length is 128 in test data - is 256 correct here?

Reply:
There was wrong field in the json string (this is taken from "length", and not "byteLength"), fixed it.